# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/ps_split_composite.glsl
# Commit: 86db847cfe57
# Full Hash: 86db847cfe570dba25289ffbe52aaaaecf76bc70
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-18 04:07:40
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 2 - Remove texture array usage from render task graph. r=nical
#   
#   With this change, all color/alpha intermediate surfaces are individual
#   textures, rather than texture arrays.
#   
# ==============================================================================

diff -r c3495de132cd -r 86db847cfe57 gfx/wr/webrender/res/ps_split_composite.glsl
--- a/gfx/wr/webrender/res/ps_split_composite.glsl	Thu Dec 17 19:53:54 2020 +0000
+++ b/gfx/wr/webrender/res/ps_split_composite.glsl	Thu Dec 17 19:54:02 2020 +0000
@@ -2,12 +2,14 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#define WR_FEATURE_TEXTURE_2D
+
 #include shared,prim_shared
 
 // interpolated UV coordinates to sample.
 varying vec2 vUv;
-// X = layer index to sample, Y = flag to allow perspective interpolation of UV.
-flat varying vec2 vLayerAndPerspective;
+// flag to allow perspective interpolation of UV.
+flat varying float vPerspective;
 flat varying vec4 vUvSampleBounds;
 
 #ifdef WR_VERTEX_SHADER
@@ -104,15 +106,15 @@
     float perspective_interpolate = float(ph.user_data.y);
 
     vUv = uv / texture_size * mix(gl_Position.w, 1.0, perspective_interpolate);
-    vLayerAndPerspective = vec2(res.layer, perspective_interpolate);
+    vPerspective = perspective_interpolate;
 }
 #endif
 
 #ifdef WR_FRAGMENT_SHADER
 void main(void) {
     float alpha = do_clip();
-    float perspective_divisor = mix(gl_FragCoord.w, 1.0, vLayerAndPerspective.y);
+    float perspective_divisor = mix(gl_FragCoord.w, 1.0, vPerspective);
     vec2 uv = clamp(vUv * perspective_divisor, vUvSampleBounds.xy, vUvSampleBounds.zw);
-    write_output(alpha * textureLod(sColor0, vec3(uv, vLayerAndPerspective.x), 0.0));
+    write_output(alpha * texture(sColor0, uv));
 }
 #endif